api/cpp/include/acquire-zarr/acquire-zarr.hh

```diff
@@ -0,0 +1,242 @@
+#ifndef ACQUIRE_ZARR_HH
+#define ACQUIRE_ZARR_HH
+
+#include <memory>
+#include <string>
+#include <vector>
+
+
+enum class AcquireZarrCompressionCodec
+{
+    COMPRESSION_NONE,
+    COMPRESSION_BLOSC_LZ4,
+    COMPRESSION_BLOSC_ZSTD,
+} ;
+
+enum class AcquireZarrDtype
+{
+    DTYPE_UNKNOWN,
+    DTYPE_UINT8,
+    DTYPE_UINT16,
+    DTYPE_INT8,
+    DTYPE_INT16,
+    DTYPE_FLOAT32,
+    DTYPE_UNIMPLEMENTED
+} ;
+
+class AcquireZarrWriter 
+{
+  public:
+    AcquireZarrWriter() { create_impl();}
+    ~AcquireZarrWriter() = default;
```

Comment:
@jeskesen I think the reason you can't use a unique pointer is that your destructor is defined here. This is equivalent to defining an inline destructor in the header file. The automatic deletion of the unique pointer happens in the destructor, but in this case, the compiler will raise an error because the size of the underlying type the pointer points to is unknown. If that's the underlying issue, the solution would be declaring the destructor in the header file and defining it in the implementation file.

```cpp
~AcquireZarrWriter(); // remove = default
```

And define it in the implementation file:
```cpp
AcquireZarrWriter::~AcquireZarrWriter() = default;
```
Try it out and let me know if it works. It’s been annoying me for some time now.